js/data/symbol_bucket.js

```diff
@@ -326,12 +338,12 @@ SymbolBucket.prototype.placeFeatures = function(collisionTile, buffers, collisio
         iconsNeedLinear: this.iconsNeedLinear
     };
 
-    var layout = this.layoutProperties;
+    var layout = this.layer.layout;
     var maxScale = collisionTile.maxScale;
 
-    elementGroups.glyph['text-size'] = layout['text-size'];
+    elementGroups.glyph['text-size'] = this.adjustedTextSize;
     elementGroups.glyph['text-font'] = layout['text-font'];
-    elementGroups.icon['icon-size'] = layout['icon-size'];
+    elementGroups.icon['icon-size'] = this.adjustedIconSize;
```

Comment:
I like how you renamed and separated adjustedIconSize from the real layout properties. Renaming it here too might be clearer: `elementGroups.icon.adjustedIconSize = this.adjustedIconSize` etc. They get used [here](https://github.com/mapbox/mapbox-gl-js/blob/70f0677d5ceeb1bd699d3b459410469d911f3483/js/render/draw_symbol.js#L149)